staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/fq_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fairqueuing
+
+import (
+	"math"
+	"sync"
+	"testing"
+	"time"
+
+	"k8s.io/apiserver/pkg/util/clock"
+)
+
+type uniformScenario []uniformClient
+
+type uniformClient struct {
+	hash          uint64
+	nThreads      int
+	nCalls        int
+	execDuration  time.Duration
+	thinkDuration time.Duration
+}
+
+// exerciseQueueSetUniformScenario.  Simple logic, only works if each
+// client's offered load is at least as large as its fair share of
+// capacity.
+func exerciseQueueSetUniformScenario(t *testing.T, qs QueueSet, sc uniformScenario, handSize int32, totalDuration time.Duration) {
```

Comment:
we can make this return a "false" on failure, and the `DummyTest` can assert that instead of failing

Reply:
yes